src/components/header/EmptyThreadIndicator.js

```diff
@@ -0,0 +1,158 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import React, { PureComponent } from 'react';
+import { withSize } from '../shared/WithSize';
+import Tooltip from '../shared/Tooltip';
+
+import type { Thread } from '../../types/profile';
+import type { Milliseconds, StartEndRange } from '../../types/units';
+
+import './EmptyThreadIndicator.css';
+
+type Props = {
+  +rangeStart: Milliseconds,
+  +rangeEnd: Milliseconds,
+  +width: number,
+  +thread: Thread,
+  +unfilteredSamplesRange: StartEndRange | null,
+};
+
+type State = {
+  hoveredItem: *,
+  mouseX: number,
+  mouseY: number,
+};
+
+class EmptyThreadIndicator extends PureComponent<Props, State> {
+  _canvas: HTMLCanvasElement | null;
+  _requestedAnimationFrame: boolean | null;
+
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      hoveredItem: null,
+      mouseX: 0,
+      mouseY: 0,
+    };
+  }
+
+  _getIndicatorPositions(): {
+    startup?: Object,
+    shutdown?: Object,
+    emptyBufferStart?: Object,
+    emptyBufferEnd?: Object,
+  } {
+    const {
+      rangeStart,
+      rangeEnd,
+      width,
+      unfilteredSamplesRange,
+      thread: {
+        processShutdownTime,
+        processStartupTime,
+        registerTime,
+        unregisterTime,
+      },
+    } = this.props;
+    const style = {};
```

Comment:
maybe you'll need to help flow here by using `const style: * = {}` so that it's not an unsealed object, but not sure. Maybe it will just work.

Reply:
Not sure what you mean here exactly.

Reply:
It's just that for Flow, `{}` is an unsealed object and has special behavior. Especially Flow doesn't warn when trying to access a non existent property. I'd like to avoid this because in this case we know the shape of this object.

Maybe it's best to always return all properties, with either `null` or a value.